include/grpc/grpc_security.h

```diff
@@ -342,12 +350,15 @@ GRPCAPI void grpc_server_credentials_release(grpc_server_credentials *creds);
      server. This parameter cannot be NULL.
    - num_key_cert_pairs indicates the number of items in the private_key_files
      and cert_chain_files parameters. It should be at least 1.
-   - force_client_auth, if set to non-zero will force the client to authenticate
+   - client_certificate_request, if set to non-zero will force the client to
+   authenticate
      with an SSL cert. Note that this option is ignored if pem_root_certs is
      NULL. */
 GRPCAPI grpc_server_credentials *grpc_ssl_server_credentials_create(
     const char *pem_root_certs, grpc_ssl_pem_key_cert_pair *pem_key_cert_pairs,
-    size_t num_key_cert_pairs, int force_client_auth, void *reserved);
+    size_t num_key_cert_pairs,
+    grpc_client_certificate_request_type client_certificate_request,
```

Comment:
enums are not type safe in C. I'm really worried that client will call this function with the old int parameter which will cause some unexpected behavior....

I would prefer to have a new function (grpc_ssl_server_credentials_create_ex) and this will avoid any confusion.